### PR TITLE
Updatingforwebsite

### PR DESCRIPTION
**Pull request by TheDuke203**
This is an update to the constants with additional stuff added it should be used alongside this 
[this update to the website](https://github.com/ps2alerts/website/pull/449)

### CODE REVIEW

File: bracket.ts [hunk 1]
```diff
@@ -1,19 +1,11 @@
 export enum Bracket {
-    UNKNOWN = -1,
-    TOTAL = 0,
-    DEAD = 1,
-    LOW = 2,
-    MEDIUM = 3,
-    HIGH = 4,
-    PRIME = 5
```
Comment (Maelstromeous) [inline bracket.ts:8]: Please revert the spacing changes and update your IDE to use the project standard of 4 spaces

File: censusOldFacilities.ts [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
-    216, // West Highlands Checkpoint
-    222, // East Canyon Checkpoint
-    223, // Abandoned NS Offices,
-    225, // Sandstone Gulch Mining
-    231, // Broken Arch Road
-    241, // NS Research Labs
-    243, // Red Ridge Communications
-    250, // Blackshard Iridium Mine
-];
+  0, // Oshur Vast Expanse
```
Comment (Maelstromeous) [inline censusOldFacilities.ts]: Spacing

File: facilityType.ts [hunk 3]
```diff
@@ -1,13 +1,26 @@
...
-    BIO_LAB = 3,
-    TECH_PLANT = 4,
-    LARGE_OUTPOST = 5,
-    SMALL_OUTPOST = 6,
-    WARPGATE = 7,
-    INTERLINK_FACILITY = 8,
-    CONSTRUCTION_OUTPOST = 9,
-    RELIC_OUTPOST = 10, // Desolation only
-    CONTAINMENT_SITE = 11
+  DEFAULT = 1, // Wat
```
Comment (Maelstromeous) [inline facilityType.ts]: Spacing

File: faction.ts [hunk 4]
```diff
@@ -1,15 +1,9 @@
 export enum Faction {
-    NONE = 0,
```
Comment (Maelstromeous) [inline faction.ts]: Spacing

File: lithafalconEndpoints.ts [hunk 5]
```diff
@@ -1,3 +1,3 @@
 export const lithafalconEndpoints = {
-    itemId: '/item',
```
Comment (Maelstromeous) [inline lithafalconEndpoints.ts:2]: Spacing

File: loadout.ts [hunk 6]
```diff
@@ -1,61 +1,133 @@
+import { Faction } from '~/constants/faction'
+
 export enum Loadout {
-    UNKNOWN = 0,
```
Comment (Maelstromeous) [inline loadout.ts:2]: Spacing

File: loadout.ts [hunk 7]
```diff
@@ -1,61 +1,133 @@
...
+  NSO_LIGHT_ASSAULT = 29,
+  NSO_MEDIC = 30,
+  NSO_ENGINEER = 31,
+  NSO_HEAVY_ASSAULT = 32,
+  NSO_MAX = 45,
 }
 
 export const loadoutArray = [
-    0,
-    1,
```
Comment (Maelstromeous) [inline loadout.ts:37]: Revert change

File: loadout.ts [hunk 8]
```diff
@@ -1,61 +1,133 @@
...
+  0, 1, 3, 4, 5, 6, 7, 8, 10, 11, 12, 13, 14, 15, 17, 18, 19, 20, 21, 28, 29,
+  30, 31, 32, 45,
+]
+export const LoadoutFaction = (loadout: Loadout | undefined): Faction => {
+  if (!loadout) {
+    return Faction.NONE
+  }
+
+  switch (loadout) {
+    case Loadout.NC_INFILTRATOR:
```
Comment (Maelstromeous) [inline loadout.ts]: Spacing

File: metagameEventState.ts [hunk 9]
```diff
@@ -1,9 +1,6 @@
 export enum MetagameEventState {
-    STARTED = 137,
```
Comment (Maelstromeous) [inline metagameEventState.ts]: Spacing

File: mqAcceptedPatterns.ts [hunk 10]
```diff
@@ -1,28 +1,28 @@
 // CAREFUL modifying this file! The values have to match what the API expects exactly or we will get
 // unprocessable messages! This should also match on the API side, so if you change this, update both.
 export enum MqAcceptedPatterns {
-    // Global Aggregates
-    GLOBAL_CHARACTER_AGGREGATE = 'globalCharacterAggregate',
```
Comment (Maelstromeous) [inline mqAcceptedPatterns.ts]: Spacing

File: playerFacility.ts [hunk 11]
```diff
@@ -1,4 +1,4 @@
 export enum PlayerFacility {
-    CAPTURE = 1,
```
Comment (Maelstromeous) [inline playerFacility.ts]: Spacing

File: ps2AlertsApiEndpoints.ts [hunk 12]
```diff
@@ -1,8 +1,9 @@
 export const ps2AlertsApiEndpoints = {
-    instanceActive: '/instances/active',
```
Comment (Maelstromeous) [inline ps2AlertsApiEndpoints.ts]: Spacing

File: ps2alertsEventState.ts [hunk 13]
```diff
@@ -1,11 +1,7 @@
 export enum Ps2alertsEventState {
-    STARTING = 0,
```
Comment (Maelstromeous) [inline ps2alertsEventState.ts]: Spacing

File: ps2alertsEventType.ts [hunk 14]
```diff
@@ -1,10 +1,7 @@
 export enum Ps2alertsEventType {
-    COMMUNITY_EVENT = 1,
```
Comment (Maelstromeous) [inline ps2alertsEventType.ts]: Spacing also I'm pretty sure this file isn't used

File: ps2alertsEventType.ts [hunk 15]
```diff
@@ -1,10 +1,7 @@
...
+  COMMUNITY_EVENT = 1,
+  JAEGER_EVENT = 2,
+  // More to add
 }
 
-export const ps2alertsEventTypeArray = [
-    1,
-    2,
-];
+export const ps2alertsEventTypeArray = [1, 2]
```
Comment (Maelstromeous) [inline ps2alertsEventType.ts:7]: Keep this change, makes sense

File: vehicle.ts [hunk 16]
```diff
@@ -1,111 +1,147 @@
+// Here's a full list: https://discord.com/channels/251073753759481856/451032574538547201/935355006071218236
+// Thanks Cooltrain!
+
 export enum Vehicle {
-    FLASH = 1,
```
Comment (Maelstromeous) [inline vehicle.ts:2]: All the spacing

File: vehicle.ts [hunk 17]
```diff
@@ -1,111 +1,147 @@
...
+  DERVISH = 2136,
+  CHIMERA = 2137,
+  DELIVERER_PROTOTYPE_2 = 2139, // Spawn ANT, UNUSED
+  LODESTAR_PROTOTYPE_2 = 2140, // Spawn Galaxy, UNUSED - NOT IN API
+  WASP_PROTOTYPE_2 = 2141, // Currently used (Cortium Valkyrie)
+  EXODUS_PPC = 2143, // Big Pew Pew on the Oshur carriers
 }
 
+// Allowed vehicles to show up in the matrix (both alert and on the stats page)
+export const allowedVehicleMatrixArray = [
```
Comment (Maelstromeous) [inline vehicle.ts]: This is website specific and should belong in that repo

File: vehicle.ts [hunk 18]
```diff
@@ -1,111 +1,147 @@
...
+  Vehicle.FLASH_XS_1,
+  Vehicle.BASTION,
+  Vehicle.FLAIL,
+  Vehicle.DELIVERER_PROTOTYPE,
+  Vehicle.WASP_PROTOTYPE,
+  Vehicle.LODESTAR_PROTOTYPE_2,
+  Vehicle.MOSQUITO_INTERCEPTOR,
+  Vehicle.SCYTHE_INTERCEPTOR,
+  Vehicle.DERVISH,
+  Vehicle.CHIMERA,
```
Comment (Maelstromeous) [inline vehicle.ts]: We also need to add the corsair here

File: vehicle.ts [hunk 19]
```diff
@@ -1,111 +1,147 @@
...
+]
+
 export const vehicleArray = [
-    1,
```
Comment (Maelstromeous) [inline vehicle.ts:58]: Revert + add corsair ID

File: world.ts [hunk 20]
```diff
@@ -1,54 +1,38 @@
 export enum World {
-    CONNERY = 1,
-    MILLER = 10,
```
Comment (Maelstromeous) [inline world.ts]: Spacing

File: world.ts [hunk 21]
```diff
@@ -1,54 +1,38 @@
...
+  MILLER = 10,
+  COBALT = 13,
+  EMERALD = 17,
+  JAEGER = 19,
+  // BRIGGS = 25,
+  SOLTECH = 40,
 }
 
 export enum WorldPS4US {
-    GENUDINE = 1000,
```
Comment (Maelstromeous) [inline world.ts]: Spacing

File: world.ts [hunk 22]
```diff
@@ -1,54 +1,38 @@
...
-    CERES = 2000,
-    // LITHCORP = 2001,
-    // RASHNU = 2002,
+  CERES = 2000,
+  // LITHCORP = 2001,
+  // RASHNU = 2002,
 }
 
-export const worldArray = [
-    1,
```
Comment (Maelstromeous) [inline world.ts]: Revert

File: world.ts [hunk 23]
```diff
@@ -1,54 +1,38 @@
...
-    17,
-    19,
-    40,
-    1000,
-    2000,
-];
+export const worldArray = [1, 10, 13, 17, 19, 40, 1000, 2000]
 
-export const pcWorldArray = [
-    1,
```
Comment (Maelstromeous) [inline world.ts]: Revert

File: zone.ts [hunk 24]
```diff
@@ -1,18 +1,52 @@
 export enum Zone {
-    INDAR = 2,
```
Comment (Maelstromeous) [inline zone.ts]: Spacing

File: zone.ts [hunk 25]
```diff
@@ -1,18 +1,52 @@
...
+  AMERISH = 6,
+  ESAMIR = 8,
+  // VR_TRAINING_NC = 96,
+  // VR_TRAINING_TR = 97,
+  // VR_TRAINING_VS = 98,
+  OSHUR = 344,
 }
 
-export const zoneArray = [
-    2,
```
Comment (Maelstromeous) [inline zone.ts]: Revert

File: zone.ts [hunk 26]
```diff
@@ -1,18 +1,52 @@
...
-    8,
-    344,
-];
+export const zoneArray = [2, 4, 6, 8, 344]
+export const zoneToWarpgateArray = new Map<Zone, number[]>([
+  [Zone.INDAR, [7801, 120000, 4801]],
+  [Zone.HOSSIN, [308000, 309000, 310000]],
+  [Zone.AMERISH, [200000, 201000, 203000]],
+  [Zone.ESAMIR, [258000, 259000, 260000]],
+  [Zone.OSHUR, [400370, 400369, 400371]],
```
Comment (Maelstromeous) [inline zone.ts]: @ryanjsims do you know if the warpgate IDs changed between the recent map changes?

File: zone.ts [hunk 27]
```diff
@@ -1,18 +1,52 @@
...
+])
+
+export const ZoneBgClassString = (zone: Zone | undefined): string => {
```
Comment (Maelstromeous) [inline zone.ts]: Website specific

File: zone.ts [hunk 28]
```diff
@@ -1,18 +1,52 @@
...
+    case Zone.AMERISH:
+      return 'bg-amerish'
+    case Zone.ESAMIR:
+      return 'bg-esamir'
+    case Zone.OSHUR:
+      return 'bg-oshur'
+  }
+}
+
+export const ZoneHexSize = (zone: Zone | undefined): number => {
```
Comment (Maelstromeous) [inline zone.ts]: Website specific

**Review by Maelstromeous — CHANGES_REQUESTED** [review]
Lots of spacing reverts to do.

File: zone.ts (hunk 27 above)
Comment (TheDuke203) [reply zone.ts]: But we have zone.ts from our submodule and then a seperate one for the website with the same name that will make it very confusing

Comment (Maelstromeous) [reply zone.ts]: We shouldn't be putting a module's specific constants in a shared module. 

Also check my comments in the website PR, you'll find your answer in there

**Comment by Maelstromeous** [comment]
Fyi for the future branch names are in dash format e.g "website-integration"

File: zone.ts (hunk 27 above)
Comment (TheDuke203) [reply zone.ts]: I agree we shouldn't be putting module specific stuff into a shared place, however the only other way would either be 
1. Make a new constants file called something different and re-route specifically those files that are looking for specifically this object to go to it.
2. Or make a new file with the same name and then re-route the ones that are looking for the object to this but in this case we know have two files in the project with the same name. Say if someone was to update this file later and they wanted to use this object again they would have to work out where it is stored in, this also applies in the 1. scenario.
So I just think that this is the lesser of the two evils.

**Pull request merged**